### PR TITLE
incomplete import

### LINKED ISSUE #1966: Framework Import causing incomplete text import if plugins are mismatch with the Framework
## Expected Behaviour
Upload should be halted and informed to the user about mismatch without importing anything in the course

## Actual Behaviour
Half buggy import completes.

## Steps to Reproduce
1. Upgrade the framework to x.x.x version
2. upload some course with a mismatch version of the x.x.x. framework 
3. Half baked import

## Versions
  - Authoring Tool Version: 0.5.0
  - Framework Version:
  - Operating System:
  - Browser:

### PR DESCRIPTION
**Pull request by lc-thomasberger**
fixes #1966 

Check if all plugins support the installed framework version before anything is imported. If not, exit import.

### CODE REVIEW

**Comment by lc-thomasberger** [comment]
Compile a list of plugins that need to be updated and present it to the user. 

Future implementation should have more fine grained options that import the content and skips the unsupported plugins.

File: frontend/src/modules/frameworkImport/views/frameworkImportView.js [hunk 1]
```diff
@@ -91,10 +91,12 @@ define(function(require){
         // TODO - add route for course destroy
```
[the remark this answers is not in the document]
Comment (lc-thomasberger) [reply frontend/src/modules/frameworkImport/views/frameworkImportView.js]: I have not looked at it too much detail.
@dancgray do you have more information about this?

[the remark this answers is not in the document]
Comment (lc-thomasberger) [reply frontend/src/modules/frameworkImport/views/frameworkImportView.js]: Thanks Dan, will remove here then.

[the remark this answers is not in the document]
Comment (taylortom) [reply frontend/src/modules/frameworkImport/views/frameworkImportView.js]: 👍

File: plugins/output/adapt/helpers.js [hunk 2]
```diff
@@ -234,6 +220,18 @@ function checkFrameworkVersion(versionMetaData, cb) {
   });
 };
 
+function checkPluginFrameworkVersion(serverFrameworkVersion, pluginMetaData) {
```
[the remark this answers is not in the document]
Comment (lc-thomasberger) [reply plugins/output/adapt/helpers.js]: done

**Comment by tomgreenfield** [comment]
I now have functional courses with custom plugins that no longer import due to absent `framework` fields.

Would people be up for relaxing the check to not fail when there's no `framework` field? Could maybe print a disclaimer, rather than error.

**Comment by lc-thomasberger** [comment]
@tomgreenfield I would not recommend this. I think we should "force" best practices for plugins. 
Alos, there is no easy way to delete plugins and no easy way to switch Framework version. 
If a user imports a plugin and it breaks compatibility with the current (or future) Framework, you are pretty much hosed. Without a framework field, we also can't check if a framework version update is possible.

**Pull request merged**